module-system/module-implementations/sov-bank/src/token.rs

```diff
@@ -103,10 +150,20 @@ impl<C: sov_modules_api::Context> Token<C> {
             None => bail!("Total supply overflow"),
         };
 
+        let mut indices = HashSet::new();
+        let mut auth_minter_list = Vec::new();
+
+        for (i, item) in authorized_minters.iter().enumerate() {
```

Comment:
hmm, so I was using the loop because I'm also deduping the source vector in case a user passes in the same address multiple times in the authorized_vector list